File: app/src/main/java/awais/instagrabber/models/HighlightModel.java
package awais.instagrabber.models;

public final class HighlightModel {
    private final String title;
    private final String id;
    private final String thumbnailUrl;

    public HighlightModel(final String title,
                          final String id,
                          final String thumbnailUrl) {
        this.title = title;
        this.id = id;
        this.thumbnailUrl = thumbnailUrl;
    }

    public String getTitle() {
        return title;
    }

    public String getId() {
        return id;
    }

    public String getThumbnailUrl() {
        return thumbnailUrl;
    }
}

File: app/src/main/java/awais/instagrabber/models/HighlightModel.kt
package awais.instagrabber.models

import awais.instagrabber.utils.TextUtils

data class HighlightModel(
    val title: String? = null,
    val id: String = "",
    val thumbnailUrl: String = "",
    val timestamp: Long = 0,
    val mediaCount: Int = 0,
) {
    val dateTime: String
        get() = TextUtils.epochSecondToString(timestamp)
}